apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -125,20 +125,19 @@ defmodule OMG.Watcher.BlockGetter do
         |> Core.ensure_block_imported_once(state)
         |> Enum.each(&DB.Transaction.update_with/1)
 
-      state = run_block_download_task(state)
```

Comment:
I would leave it on the top, you can download new blocks without checking the current one